Sources/Rendering/Core/ImageResliceMapper/test/testImageResliceMapperBorderRendering.js

```diff
@@ -0,0 +1,131 @@
+import test from 'tape';
+import testUtils from 'vtk.js/Sources/Testing/testUtils';
+
+import 'vtk.js/Sources/Rendering/OpenGL/Profiles/All';
+
+import { SlabTypes } from 'vtk.js/Sources/Rendering/Core/ImageResliceMapper/Constants';
+import vtkColorTransferFunction from 'vtk.js/Sources/Rendering/Core/ColorTransferFunction';
+import vtkDataArray from 'vtk.js/Sources/Common/Core/DataArray';
+import vtkImageData from 'vtk.js/Sources/Common/DataModel/ImageData';
+import vtkImageProperty from 'vtk.js/Sources/Rendering/Core/ImageProperty';
+import vtkImageResliceMapper from 'vtk.js/Sources/Rendering/Core/ImageResliceMapper';
+import vtkImageSlice from 'vtk.js/Sources/Rendering/Core/ImageSlice';
+import vtkPiecewiseFunction from 'vtk.js/Sources/Common/DataModel/PiecewiseFunction';
+import vtkPlane from 'vtk.js/Sources/Common/DataModel/Plane';
+import vtkRenderer from 'vtk.js/Sources/Rendering/Core/Renderer';
+import vtkRenderWindow from 'vtk.js/Sources/Rendering/Core/RenderWindow';
+
+import baseline from './testImageResliceMapperBorderRendering.png';
+
+test.onlyIfWebGL('Test ImageResliceMapper', (t) => {
+  const gc = testUtils.createGarbageCollector(t);
+  t.ok(
+    'rendering',
+    'vtkImageResliceMapper testImageResliceMapperBorderRendering'
+  );
+
+  const renderWindow = gc.registerResource(vtkRenderWindow.newInstance());
+  const renderer = gc.registerResource(vtkRenderer.newInstance());
+  renderWindow.addRenderer(renderer);
+
+  // Opaque green for the background
+  renderer.setBackground(0, 1, 0);
+
+  const imageDimension = 4;
+
+  const ppty = gc.registerResource(vtkImageProperty.newInstance());
+  const mapper = gc.registerResource(vtkImageResliceMapper.newInstance());
+  const slicePlane = gc.registerResource(vtkPlane.newInstance());
+  slicePlane.setNormal(0, 1, 0);
+  mapper.setSlicePlane(slicePlane);
+  mapper.setSlabType(SlabTypes.MAX);
+
+  // Opaque red for out of volume rendering
+  mapper.setBackgroundColor(1, 0, 0, 1);
+
+  const actor = gc.registerResource(vtkImageSlice.newInstance());
+  actor.setMapper(mapper);
+  renderer.addActor(actor);
+
+  const minScalar = 0;
+  const maxScalar = 2 * imageDimension;
+
+  ppty.setColorWindow(maxScalar - minScalar);
+  ppty.setColorLevel(0.5 * (maxScalar + minScalar));
+
+  const rgb = gc.registerResource(vtkColorTransferFunction.newInstance());
+  rgb.addRGBPoint(minScalar, 0, 0, 0);
+  rgb.addRGBPoint(maxScalar, 1, 1, 1);
+  ppty.setRGBTransferFunction(rgb);
+
+  const ofun = gc.registerResource(vtkPiecewiseFunction.newInstance());
+  ofun.addPoint(minScalar, 1);
+  ofun.addPoint(maxScalar, 1);
+  ppty.setPiecewiseFunction(ofun);
+
+  ppty.setInterpolationTypeToNearest();
+  actor.setProperty(ppty);
+
+  const imageData = gc.registerResource(vtkImageData.newInstance());
+
+  imageData.setOrigin(0, 0, 0);
+  imageData.setSpacing(1, 1, 1);
```

Comment:
it would be great to have a non 1 spacing (in case the 0.5 offset is spacing-related)